[Bill-1] iP

Bill-1 opened the pull request:
# Ujin

> "The mind is not a vessel to be filled, but a fire to be kindled." - [Plutarch](https://en.wikipedia.org/wiki/Plutarch)

Ujin frees your mind of having to remember things you need to do. It's:
- **text-based**
- **easy to learn**
- ~~FAST~~ **SUPER FAST** to use

All you need to do is:
1. [download it from here](#)
2. double-click it
3. add your tasks
4. let it manage your tasks for you

And it is **FREE**!

## Features
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    Application.launch(MainApp.class, args);
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Bill-1 committed:
Level 0. Rename, Greet, Exit

Bill-1 committed:
Level 1. Echo

Bill-1 committed:
Level 2. Add, List

Bill-1 committed:
Level 3. Mark as Done

Bill-1 committed:
Level 4. ToDos, Events, Deadlines

Bill-1 committed:
A-TextUiTesting

Bill-1 committed:
Level 5. Handle Errors

Bill-1 committed:
Level 6. Delete

Bill-1 committed:
Level 7. Save

Bill-1 committed:
Level 8. Dates and Times

Bill-1 committed:
A-MoreOOP

Bill-1 committed:
A-Packages

Bill-1 committed:
Convert to the standard convention

Bill-1 committed:
Merge branch 'add-gradle-support'

Bill-1 committed:
Add JUnit tests

Bill-1 committed:
Add JavaDoc to all the methods and classes

Bill-1 committed:
Add find command

Bill-1 committed:
Merge tag 'Level-9'

Bill-1 committed:
Add GUI and fix style

xinniyee commented on src/main/java/ujin/helper/Parser.java:
I like how you made use of helper methods

xinniyee commented on src/main/java/ujin/ui/Ui.java:
I like how you named your methods such that it is easy to understand! 👍

xinniyee commented on src/main/java/ujin/task/TaskList.java:
```diff
@@ -0,0 +1,67 @@
+     *
+     * @return The number of tasks in the task list.
+     */
+    public int size() {
```
I didn't like how this method is named. Perhaps there could a verb could be used to make it clearer?

xinniyee commented on src/main/java/ujin/command/FindCommand.java:
```diff
@@ -0,0 +1,46 @@
+public class FindCommand extends Command {
+
+    private final String KEYWORD;
+    private final TaskList MATCHLIST = new TaskList();
```
Maybe you should use camel case for variables?

xinniyee reviewed (commented):
LGTM! Just some small changes. Great!

Bill-1 committed:
Add assertions to verify program invariants

- Added assertions in the execute() method in FindCommand to ensure size is not negative
- This assertion help catch logic error but never will

Bill-1 committed:
Add abstraction on formatting the date

- Added formatting the DateLocaltime variable in Task class.
- Added shortcut in child classes Deadline and Event.

Bill-1 committed:
Add stream on UI

Bill-1 committed:
Merge pull request #1 from Bill-1/branch-A-Assertions

Add assertions to verify program invariants

Bill-1 committed:
Merge branch 'master' into branch-A-CodeQuality

Bill-1 committed:
Merge branch 'master' into branch-A-Streams

Bill-1 committed:
Merge pull request #2 from Bill-1/branch-A-CodeQuality

Add abstraction on formatting the date

Bill-1 committed:
Merge pull request #3 from Bill-1/branch-A-Streams

Add stream on UI

Bill-1 committed:
Change the user image

Bill-1 committed:
Add contact, not finished

patil-np commented on src/main/java/ujin/ui/Ui.java:
```diff
@@ -0,0 +1,169 @@
+     * @param newTask The task that was added.
+     * @param size The current size of the task list.
+     */
+    public String addedTask(Task newTask, int size) {
```
This is just a nitpick but perhaps this method carrying out an action can be named starting with a verb e.g. `confirmTaskAddedMessage`!

patil-np commented on src/main/java/ujin/Contact.java:
```diff
@@ -0,0 +1,39 @@
+package ujin;
+
+public class Contact {
```
I like how you used nouns for things and verbs for actions in this class!

patil-np reviewed (commented):
LGTM. Good job following the coding conventions!

Bill-1 committed:
Add latest change

Bill-1 committed:
Update README.md

Bill-1 committed:
Update README.md

Bill-1 committed:
Update README.md

Bill-1 committed:
Add GUI window image

Bill-1 committed:
Finalize it

Bill-1 committed:
Update README.md

Bill-1 committed:
Fix user guide

Bill-1 committed:
Update README.md

Bill-1 committed:
Update README.md
